pkg/controllers/disruption/consolidation_test.go

```diff
@@ -608,69 +623,112 @@ var _ = Describe("Consolidation", func() {
+				var wg sync.WaitGroup
+				ExpectTriggerVerifyAction(&wg)
+				ExpectMakeNewNodeClaimsReady(ctx, env.Client, &wg, cluster, cloudProvider, 1)
+				ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKey{})
+				wg.Wait()
 
-			// Process the item so that the nodes can be deleted.
-			ExpectReconcileSucceeded(ctx, queue, types.NamespacedName{})
+				// Process the item so that the nodes can be deleted.
+				ExpectReconcileSucceeded(ctx, queue, types.NamespacedName{})
 
-			// Cascade any deletion of the nodeclaim to the node
-			ExpectNodeClaimsCascadeDeletion(ctx, env.Client, nodeClaim)
+				// Cascade any deletion of the nodeclaim to the node
+				ExpectNodeClaimsCascadeDeletion(ctx, env.Client, nodeClaim)
 
-			// should create a new nodeclaim as there is a cheaper one that can hold the pod
-			nodeClaims := ExpectNodeClaims(ctx, env.Client)
-			nodes := ExpectNodes(ctx, env.Client)
-			Expect(nodeClaims).To(HaveLen(1))
-			Expect(nodes).To(HaveLen(1))
+				// should create a new nodeclaim as there is a cheaper one that can hold the pod
+				nodeClaims := ExpectNodeClaims(ctx, env.Client)
+				nodes := ExpectNodes(ctx, env.Client)
+				Expect(nodeClaims).To(HaveLen(1))
+				Expect(nodes).To(HaveLen(1))
 
-			// Expect that the new nodeclaim does not request the most expensive instance type
-			Expect(nodeClaims[0].Name).ToNot(Equal(nodeClaim.Name))
-			Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Has(v1.LabelInstanceTypeStable)).To(BeTrue())
-			Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Get(v1.LabelInstanceTypeStable).Has(mostExpensiveInstance.Name)).To(BeFalse())
+				// Expect that the new nodeclaim does not request the most expensive instance type
+				Expect(nodeClaims[0].Name).ToNot(Equal(nodeClaim.Name))
+				Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Has(v1.LabelInstanceTypeStable)).To(BeTrue())
+				Expect(scheduling.NewNodeSelectorRequirements(nodeClaims[0].Spec.Requirements...).Get(v1.LabelInstanceTypeStable).Has(mostExpensiveInstance.Name)).To(BeFalse())
 
-			// and delete the old one
-			ExpectNotFound(ctx, env.Client, nodeClaim, node)
-		})
-		It("can replace nodes if another nodePool returns no instance types", func() {
+				// and delete the old one
+				ExpectNotFound(ctx, env.Client, nodeClaim, node)
+			},
+			Entry("if the candidate is on-demand node", false),
+			Entry("if the candidate is spot node", true),
+		)
+		It("cannot replace spot node if less than minimum InstanceTypes flexibility", func() {
+			// Forcefully shrink the possible instanceTypes to be lower than 15 to replace a nodeclaim
+			cloudProvider.InstanceTypes = lo.Slice(fake.InstanceTypesAssorted(), 0, 5)
+			// Forcefully assign lowest possible instancePrice to make sure we have atleast one instance
+			// that is lower than the current node.
+			cloudProvider.InstanceTypes[0].Offerings[0].Price = 0.001
```

Comment:
nit: I don't think there's a check to see if this is non-zero, so why not just make it 0?

Reply:
Usually prices are not 0 in value and I thought it might get caught in validation somewhere. So, was realistic with initializing it with very low value.